Neo4j.Driver/Neo4j.Driver/Internal/IO/BoltReader.cs

```diff
@@ -81,14 +81,22 @@ public Task ReadAsync(IMessageResponseHandler responseHandler)
                 _chunkReader.ReadNextMessageAsync(_bufferStream)
                     .ContinueWith(t =>
                     {
-                        ProcessMessage(responseHandler);
+                        ConsumeMessages(responseHandler);
                     }, TaskContinuationOptions.ExecuteSynchronously);
         }
 
-        private void ProcessMessage(IMessageResponseHandler responseHandler)
+        private void ConsumeMessages(IMessageResponseHandler responseHandler)
         {
             _bufferStream.Position = 0;
 
+            while (_bufferStream.Length > _bufferStream.Position)
```

Comment:
_chunkReader reads a finite number of bytes into this _bufferStream and its Length does not change in ConsumeMessages. However, while we're reading messages from it, we are advancing its Position. This loop ensures that all messages that are read in one ReadNextMessage(Async) call has been consumed.